src/Component/Symfony/Routing/Factory/RouteFactoryInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Symfony\Routing\Factory;
+
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+interface RouteFactoryInterface
+{
+    /**
+     * @return RouteCollection
+     */
+    public function createRouteCollection();
```

Comment:
```suggestion
    public function createRouteCollection(): RouteCollection;
```

Old interface didn't have this, but does this PR allows this change?

Reply:
hum this is a bc-break.

Reply:
Yes, but will anybody notice in reality? I feel, like UPGRADE mention is enough, though it can be left for 2.0.

Reply:
Extracted to https://github.com/Sylius/SyliusResourceBundle/issues/754